core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -535,223 +730,335 @@ func TestLogPoller_SynchronizedWithGeth(t *testing.T) {
 
 func TestLogPoller_PollAndSaveLogs(t *testing.T) {
 	t.Parallel()
-	th := SetupTH(t, 2, 3, 2)
 
-	// Set up a log poller listening for log emitter logs.
-	err := th.LogPoller.RegisterFilter(logpoller.Filter{
-		"Test Emitter 1 & 2", []common.Hash{EmitterABI.Events["Log1"].ID, EmitterABI.Events["Log2"].ID},
-		[]common.Address{th.EmitterAddress1, th.EmitterAddress2}, 0,
-	})
-	require.NoError(t, err)
+	tests := []struct {
+		name          string
+		finalityDepth int64
+		finalityTag   bool
+	}{
+		{
+			name:          "fixed finality depth without finality tag",
+			finalityDepth: 3,
+			finalityTag:   false,
+		},
+		{
+			name:          "chain finality in use",
+			finalityDepth: 0,
+			finalityTag:   true,
+		},
+	}
 
-	b, err := th.Client.BlockByNumber(testutils.Context(t), nil)
-	require.NoError(t, err)
-	require.Equal(t, uint64(1), b.NumberU64())
-	require.Equal(t, uint64(10), b.Time())
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			th := SetupTH(t, tt.finalityTag, tt.finalityDepth, 3, 2)
```

Comment:
nit: could pull this into its own function to reduce nesting, ditto for poll and save logs test